device/bgpv4.yaml

```diff
@@ -8,59 +8,134 @@ components:
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         dut_as_number_2_byte:
+          description: >-
+            Autonomous system (AS) number of the BGP peer router (DUT)
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         as_number_4_byte:
+          description: >-
+            Autonomous system (AS) number of 4 byte
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         as_number_set_mode:
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         as_type:
-          description: >- 
-            The type of BGP autonomous system. 
-            External BGP (EBGP) is used for BGP links between two or more 
+          description: >-
+            The type of BGP autonomous system.
+            External BGP (EBGP) is used for BGP links between two or more
             autonomous systems.
             Internal BGP (IBGP) is used within a single autonomous system.
           type: string
           enum: [IBGP, EBGP]
         hold_time_interval:
+          description: >-
+            Number of seconds the sender proposes for the value of the Hold Timer
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         keep_alive_interval:
+          description: >-
+            Number of seconds between transmissions of Keep Alive messages by router
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         graceful_restart:
-          $ref: './patterns.yaml#/components/schemas/Device.Pattern'
+          description: >-
+            Graceful Restart allows the BGP forwarding state to be preserved during a BGP restart
+          type: boolean
         authentication:
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         ttl:
+          description: >-
+            The limited number of iterations that a unit of data can experience before the data is discarded.
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
-        dut_ipv4_address:         
+        dut_ipv4_address:
+          description: >-
+            IPv4 address of the BGP peer for the session
           $ref: './patterns.yaml#/components/schemas/Device.Pattern'
         ipv4:
           $ref: './ipv4.yaml#/components/schemas/Device.Ipv4'
+        bgpv4_route_range:
+          description: >-
+            Emulated BGPv4 route range
+          type: array
+          items:
+            $ref: '#/components/schemas/Device.Bgpv4RouteRange'
+        bgpv6_route_range:
+          description: >-
+            Emulated bgpv6 route range
+          type: array
+          items:
+            $ref: '#/components/schemas/Device.Bgpv6RouteRange'
 
     Device.Bgpv4RouteRange:
       description: >-
-        Emulated bgpv4 route range
+        Emulated BGPv4 route range
       type: object
       required: [name]
       properties:
         name: '../common/common.yaml#/components/x-inline/UniqueName'
+        route_count:
```

Comment:
@ajbalogh @ankur-sheth pondering if we actually need route count at all?  can't that be expressed as a pattern off of say the address itself?  i'm not sure what makes sense or would make it easier for other bgp implementations where it doesn't have bgp session range or route range defined?